mobile/examples/cc/fetch_client/fetch_client.cc

```diff
@@ -55,45 +58,39 @@ void Fetch::sendRequest(const absl::string_view url_string) {
     absl::MutexLock lock(&engine_mutex_);
     stream_prototype = engine_->streamClient()->newStreamPrototype();
   }
-  stream_prototype->setOnHeaders(
-      [](Platform::ResponseHeadersSharedPtr headers, bool, envoy_stream_intel intel) {
-        std::cerr << "Received headers on connection: " << intel.connection_id << "\n";
-
-        const Platform::RawHeaderMap raw_headers = headers->allHeaders();
-        for (const auto& [key, header] : raw_headers) {
-          for (const auto& val : header) {
-            std::cout << key << ": " << val << "\n";
-          }
-        }
-      });
-  stream_prototype->setOnData([](envoy_data c_data, bool fin) {
-    std::cout << Data::Utility::copyToString(c_data);
-    if (fin) {
-      std::cout << "Received final data\n";
+  EnvoyStreamCallbacks stream_callbacks;
+  stream_callbacks.on_headers_ = [&](const Http::ResponseHeaderMap& headers, bool /* end_stream */,
+                                     envoy_stream_intel intel) {
+    std::cerr << "Received headers on connection: " << intel.connection_id << "with headers:\n"
+              << headers << "\n";
+  };
+  stream_callbacks.on_data_ = [&](const Buffer::Instance& buffer, uint64_t length, bool end_stream,
+                                  envoy_stream_intel) {
+    std::string response_body(length, ' ');
+    buffer.copyOut(0, length, response_body.data());
+    std::cerr << response_body << "\n";
+    if (end_stream) {
+      std::cerr << "Received final data\n";
     }
-    release_envoy_data(c_data);
-  });
-  stream_prototype->setOnComplete(
-      [&request_finished](envoy_stream_intel, envoy_final_stream_intel final_intel) {
-        std::cerr << "Request finished after "
-                  << final_intel.stream_end_ms - final_intel.stream_start_ms << "ms\n";
-        request_finished.Notify();
-      });
-  stream_prototype->setOnError([&request_finished](Platform::EnvoyErrorSharedPtr,
-                                                   envoy_stream_intel,
-                                                   envoy_final_stream_intel final_intel) {
+  };
+  stream_callbacks.on_complete_ = [&](envoy_stream_intel, envoy_final_stream_intel final_intel) {
```

Comment:
this and below can just capture `[&request_finished]`, right?

Reply:
Yep. Updated.